tests/platform_tests/api/test_fan.py

```diff
@@ -0,0 +1,200 @@
+        self.assert_expectations()
+
+    def test_get_model(self, duthost, localhost, platform_api_conn):
+        for i in range(self.num_fans):
+            model = fan.get_model(platform_api_conn, i)
+
+            if self.expect(model is not None, "Unable to retrieve fan {} model".format(i)):
+                self.expect(isinstance(model, str),
+                            "Fan {} model appears incorrect".format(i))
+        self.assert_expectations()
+
+    def test_get_serial(self, duthost, localhost, platform_api_conn):
+        for i in range(self.num_fans):
+            serial = fan.get_serial(platform_api_conn, i)
+            if self.expect(serial is not None, "Unable to retrieve fan {} serial number".format(i)):
+                self.expect(isinstance(serial, str),
+                            "Fan {} serial number appears incorrect".format(i))
+        self.assert_expectations()
+
+    def test_get_status(self, duthost, localhost, platform_api_conn):
+        for i in range(self.num_fans):
+            status = fan.get_status(platform_api_conn, i)
+            if self.expect(status is not None, "Unable to retrieve fan {} status".format(i)):
+                self.expect(isinstance(status, bool),
+                            "Fan {} status appears incorrect".format(i))
+        self.assert_expectations()
+
+    #
+    # Functions to test methods defined in FanBase class
+    #
+
+    def test_get_speed(self, duthost, localhost, platform_api_conn):
+        # Ensure the fan speed is sane
+        for i in range(self.num_fans):
+            speed = fan.get_speed(platform_api_conn, i)
+            if self.expect(speed is not None, "Unable to retrieve Fan {} speed".format(i)):
+                if self.expect(isinstance(speed, int), "Fan {} speed appears incorrect".format(i)):
+                    self.expect(speed > 0 and speed <= 100,
+                                "Fan {} speed {} reading is not within range".format(i, speed))
+        self.assert_expectations()
+
+    def test_get_direction(self, duthost, localhost, platform_api_conn):
+        # Ensure the fan speed is sane
+        FAN_DIRECTION_LIST = [
+            "intake",
+            "exhaust",
+            "N/A",
+        ]
+        for i in range(self.num_fans):
+            direction = fan.get_direction(platform_api_conn, i)
+            if self.expect(direction is not None, "Unable to retrieve Fan {} direction".format(i)):
+                self.expect(direction in FAN_DIRECTION_LIST,
+                            "Fan {} direction is not one of predefined directions".format(i))
+
+        self.assert_expectations()
+
+    def test_get_fans_target_speed(self, duthost, localhost, platform_api_conn):
+
+        for i in range(self.num_fans):
+            fan_instance = chassis.get_fan(platform_api_conn, i)
+            self.expect(fan_instance and fan_instance ==
+                        fan_list[i], "Fan {} is incorrect".format(i))
+
```

Comment:
As mentioned previously, this consistency check is done in the chassis tests. It is redundant here. Remove.

Reply:
fixed